.github/workflows/build-and-deploy-react-app.yaml

```diff
@@ -43,26 +43,31 @@ jobs:
           role-session-name: github-pipeline-session
           aws-region: 'eu-west-2'
           audience: sts.amazonaws.com
-      - name: Find S3 Bucket
-        id: bucket
-        run: |
-          echo "${{ github.repository }}"
-          echo "${{ github.environment }}"
-          echo "${{ inputs.deployment_workspace }}"
-      - name: Generate Bucket Name
-        id: generate-bucket
+
+      - name: Handle Deployment Workspace and Bucket Name
         run: |
+          if [[ "${GITHUB_REF##*/}" == "main" ]]; then
+            echo "deployment_workspace=main" >> $GITHUB_ENV
+          else
+            echo "deployment_workspace=$(awk -F- '{print $1}' <<< "${GITHUB_REF##*/}")-$(awk -F- '{print $2}' <<< "${GITHUB_REF##*/}")" >> $GITHUB_ENV
+          fi
+
           REPO_NAME=$(echo "${{ github.repository }}" | awk -F/ '{print $2}')
           WORKSPACE=${{ inputs.deployment_workspace }}
-
           ENVIRONMENT=${{ inputs.github_environment }}
 
-          BUCKET_NAME="nhse-${REPO_NAME}-${ENVIRONMENT}-front-end-${WORKSPACE}"
-
+          if [[ "${WORKSPACE}" == "main" ]]; then
```

Comment:
Again , doesn't workspace get set to default if the branch is main?

Reply:
that's true, i can refactor the if statement to used default when the branch is main